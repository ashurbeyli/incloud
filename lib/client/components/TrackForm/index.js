import React from 'react';
import { Button, ControlLabel, FormControl, FormGroup, Modal } from 'react-bootstrap';

export const getValidationState = (task) => {
  const length = task.description.length;
  if (length > 5) return 'success';
  else if (length > 0) return 'error';
  return null;
};

const TrackForm = ({ handleChange, onClose, onSubmit, show, task }) => {

  return (
    <div className="static-modal">
      <Modal show={show}>
        <Modal.Header>
          <Modal.Title>New Task</Modal.Title>
        </Modal.Header>

        <Modal.Body>
          <form>
            <FormGroup
              controlId="taskName"
              validationState={getValidationState(task)}
            >
              <ControlLabel>Task Description</ControlLabel>
              <FormControl
                type="text"
                value={task.description}
                placeholder="Enter description"
                onChange={(e) => handleChange(e.target.value)}
              />
              <FormControl.Feedback/>
            </FormGroup>
          </form>
        </Modal.Body>

        <Modal.Footer>
          <Button type="button" onClick={onClose}>Close</Button>
          <Button bsStyle="primary" type="submit" onClick={onSubmit}>Save</Button>
        </Modal.Footer>
      </Modal>
    </div>
  );

};

export default TrackForm;