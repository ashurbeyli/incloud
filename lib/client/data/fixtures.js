export const dumbTask = {
  description: '',
  pieces: [],
};

export const task = {
  createdAt: '2018-04-19T22:07:34.639Z',
  description: 'asdfafa',
  pieces: [],
  updatedAt: '2018-04-19T22:07:34.640Z',
  user_id: '5ad65aa44212e29c4a922af7',
  __v: 0,
  _id: '5ad91326115be0385d527a1b'
};

export const tasks = [
  task,
  {
    createdAt: '2018-04-19T22:07:34.639Z',
    description: 'asdfafa',
    pieces: [],
    updatedAt: '2018-04-19T22:07:34.640Z',
    user_id: '5ad65aa44212e29c4a922af7',
    __v: 0,
    _id: '5ad916889a0c1f39b919eff7'
  },
];