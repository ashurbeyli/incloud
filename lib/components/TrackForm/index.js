import React from 'react';
import { Button, ControlLabel, FormControl, FormGroup, HelpBlock, Modal } from 'react-bootstrap';

import TasksApi from '../../api-client/tasksApi';

export default class TrackForm extends React.Component {

  constructor(props) {
    super(props);
  }

  getValidationState = () => {
    const length = this.props.task.description.length;
    if (length > 5) return 'success';
    else if (length > 0) return 'error';
    return null;
  }

  render() {
    return (
      <div className="static-modal">
        <Modal show={this.props.show}>
          <Modal.Header>
            <Modal.Title>New Task</Modal.Title>
          </Modal.Header>

          <Modal.Body>
            <form>
              <FormGroup
                controlId="taskName"
                validationState={this.getValidationState()}
              >
                <ControlLabel>Task Description</ControlLabel>
                <FormControl
                  type="text"
                  value={this.props.task.description}
                  placeholder="Enter description"
                  onChange={e => this.props.handleChange(e.target.value)}
                />
                <FormControl.Feedback/>
              </FormGroup>
            </form>
          </Modal.Body>

          <Modal.Footer>
            <Button type="button" onClick={this.props.onClose}>Close</Button>
            <Button bsStyle="primary" type="submit" onClick={this.props.onSubmit}>Save</Button>
          </Modal.Footer>
        </Modal>
      </div>
    );
  }

}