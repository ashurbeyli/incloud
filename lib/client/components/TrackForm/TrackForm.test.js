import React from 'react';
import TrackForm, { getValidationState } from './';

import { dumbTask } from '../../data/fixtures';

const props = {
  handleChange: () => {},
  onClose: () => {},
  onSubmit: () => {},
  show: true,
  task: dumbTask
};


describe('TrackForm', () => {
  const wrapper = shallow(<TrackForm {...props} />);

  it('should contain Modal', () => {
    expect(wrapper.find('Modal').exists()).toBe(true);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('getValidationState()', () => {
    it('should return "null" from getValidationState()', () => {
      expect(getValidationState(props.task)).toEqual(null);
    });

    it('should return "error" from getValidationState()', () => {
      props.task.description = 'asd';
      expect(getValidationState(props.task)).toEqual('error');
    });

    it('should return "error" from getValidationState()', () => {
      props.task.description = 'this is more than 5';
      expect(getValidationState(props.task)).toEqual('success');
    });
  });

});