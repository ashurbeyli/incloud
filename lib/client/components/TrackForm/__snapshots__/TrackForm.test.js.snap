// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackForm renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TrackForm
    handleChange={[Function]}
    onClose={[Function]}
    onSubmit={[Function]}
    show={true}
    task={
      Object {
        "description": "",
        "pieces": Array [],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Modal
        animation={true}
        autoFocus={true}
        backdrop={true}
        bsClass="modal"
        dialogComponentClass={[Function]}
        enforceFocus={true}
        keyboard={true}
        manager={
          ModalManager {
            "add": [Function],
            "containers": Array [],
            "data": Array [],
            "handleContainerOverflow": true,
            "hideSiblingNodes": true,
            "isTopModal": [Function],
            "modals": Array [],
            "remove": [Function],
          }
        }
        onHide={[Function]}
        renderBackdrop={[Function]}
        restoreFocus={true}
        show={true}
      >
        <ModalHeader
          bsClass="modal-header"
          closeButton={false}
          closeLabel="Close"
        >
          <ModalTitle
            bsClass="modal-title"
            componentClass="h4"
          >
            New Task
          </ModalTitle>
        </ModalHeader>
        <ModalBody
          bsClass="modal-body"
          componentClass="div"
        >
          <form>
            <FormGroup
              bsClass="form-group"
              controlId="taskName"
              validationState={null}
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Task Description
              </ControlLabel>
              <FormControl
                bsClass="form-control"
                componentClass="input"
                onChange={[Function]}
                placeholder="Enter description"
                type="text"
                value=""
              />
              <FormControlFeedback
                bsClass="form-control-feedback"
                bsRole="feedback"
              />
            </FormGroup>
          </form>
        </ModalBody>
        <ModalFooter
          bsClass="modal-footer"
          componentClass="div"
        >
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Close
          </Button>
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="primary"
            disabled={false}
            onClick={[Function]}
            type="submit"
          >
            Save
          </Button>
        </ModalFooter>
      </Modal>,
      "className": "static-modal",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animation": true,
        "autoFocus": true,
        "backdrop": true,
        "bsClass": "modal",
        "children": Array [
          <ModalHeader
            bsClass="modal-header"
            closeButton={false}
            closeLabel="Close"
          >
            <ModalTitle
              bsClass="modal-title"
              componentClass="h4"
            >
              New Task
            </ModalTitle>
          </ModalHeader>,
          <ModalBody
            bsClass="modal-body"
            componentClass="div"
          >
            <form>
              <FormGroup
                bsClass="form-group"
                controlId="taskName"
                validationState={null}
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Task Description
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  onChange={[Function]}
                  placeholder="Enter description"
                  type="text"
                  value=""
                />
                <FormControlFeedback
                  bsClass="form-control-feedback"
                  bsRole="feedback"
                />
              </FormGroup>
            </form>
          </ModalBody>,
          <ModalFooter
            bsClass="modal-footer"
            componentClass="div"
          >
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="default"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Close
            </Button>
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="primary"
              disabled={false}
              onClick={[Function]}
              type="submit"
            >
              Save
            </Button>
          </ModalFooter>,
        ],
        "dialogComponentClass": [Function],
        "enforceFocus": true,
        "keyboard": true,
        "manager": ModalManager {
          "add": [Function],
          "containers": Array [],
          "data": Array [],
          "handleContainerOverflow": true,
          "hideSiblingNodes": true,
          "isTopModal": [Function],
          "modals": Array [],
          "remove": [Function],
        },
        "onHide": [Function],
        "renderBackdrop": [Function],
        "restoreFocus": true,
        "show": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "modal-header",
            "children": <ModalTitle
              bsClass="modal-title"
              componentClass="h4"
            >
              New Task
            </ModalTitle>,
            "closeButton": false,
            "closeLabel": "Close",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "modal-title",
              "children": "New Task",
              "componentClass": "h4",
            },
            "ref": null,
            "rendered": "New Task",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "modal-body",
            "children": <form>
              <FormGroup
                bsClass="form-group"
                controlId="taskName"
                validationState={null}
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Task Description
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  onChange={[Function]}
                  placeholder="Enter description"
                  type="text"
                  value=""
                />
                <FormControlFeedback
                  bsClass="form-control-feedback"
                  bsRole="feedback"
                />
              </FormGroup>
            </form>,
            "componentClass": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FormGroup
                bsClass="form-group"
                controlId="taskName"
                validationState={null}
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Task Description
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  onChange={[Function]}
                  placeholder="Enter description"
                  type="text"
                  value=""
                />
                <FormControlFeedback
                  bsClass="form-control-feedback"
                  bsRole="feedback"
                />
              </FormGroup>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-group",
                "children": Array [
                  <ControlLabel
                    bsClass="control-label"
                    srOnly={false}
                  >
                    Task Description
                  </ControlLabel>,
                  <FormControl
                    bsClass="form-control"
                    componentClass="input"
                    onChange={[Function]}
                    placeholder="Enter description"
                    type="text"
                    value=""
                  />,
                  <FormControlFeedback
                    bsClass="form-control-feedback"
                    bsRole="feedback"
                  />,
                ],
                "controlId": "taskName",
                "validationState": null,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "control-label",
                    "children": "Task Description",
                    "srOnly": false,
                  },
                  "ref": null,
                  "rendered": "Task Description",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "form-control",
                    "componentClass": "input",
                    "onChange": [Function],
                    "placeholder": "Enter description",
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "form-control-feedback",
                    "bsRole": "feedback",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": "form",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "modal-footer",
            "children": Array [
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Close
              </Button>,
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="primary"
                disabled={false}
                onClick={[Function]}
                type="submit"
              >
                Save
              </Button>,
            ],
            "componentClass": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "block": false,
                "bsClass": "btn",
                "bsStyle": "default",
                "children": "Close",
                "disabled": false,
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": "Close",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "block": false,
                "bsClass": "btn",
                "bsStyle": "primary",
                "children": "Save",
                "disabled": false,
                "onClick": [Function],
                "type": "submit",
              },
              "ref": null,
              "rendered": "Save",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Modal
          animation={true}
          autoFocus={true}
          backdrop={true}
          bsClass="modal"
          dialogComponentClass={[Function]}
          enforceFocus={true}
          keyboard={true}
          manager={
            ModalManager {
              "add": [Function],
              "containers": Array [],
              "data": Array [],
              "handleContainerOverflow": true,
              "hideSiblingNodes": true,
              "isTopModal": [Function],
              "modals": Array [],
              "remove": [Function],
            }
          }
          onHide={[Function]}
          renderBackdrop={[Function]}
          restoreFocus={true}
          show={true}
        >
          <ModalHeader
            bsClass="modal-header"
            closeButton={false}
            closeLabel="Close"
          >
            <ModalTitle
              bsClass="modal-title"
              componentClass="h4"
            >
              New Task
            </ModalTitle>
          </ModalHeader>
          <ModalBody
            bsClass="modal-body"
            componentClass="div"
          >
            <form>
              <FormGroup
                bsClass="form-group"
                controlId="taskName"
                validationState={null}
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Task Description
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  onChange={[Function]}
                  placeholder="Enter description"
                  type="text"
                  value=""
                />
                <FormControlFeedback
                  bsClass="form-control-feedback"
                  bsRole="feedback"
                />
              </FormGroup>
            </form>
          </ModalBody>
          <ModalFooter
            bsClass="modal-footer"
            componentClass="div"
          >
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="default"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Close
            </Button>
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="primary"
              disabled={false}
              onClick={[Function]}
              type="submit"
            >
              Save
            </Button>
          </ModalFooter>
        </Modal>,
        "className": "static-modal",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animation": true,
          "autoFocus": true,
          "backdrop": true,
          "bsClass": "modal",
          "children": Array [
            <ModalHeader
              bsClass="modal-header"
              closeButton={false}
              closeLabel="Close"
            >
              <ModalTitle
                bsClass="modal-title"
                componentClass="h4"
              >
                New Task
              </ModalTitle>
            </ModalHeader>,
            <ModalBody
              bsClass="modal-body"
              componentClass="div"
            >
              <form>
                <FormGroup
                  bsClass="form-group"
                  controlId="taskName"
                  validationState={null}
                >
                  <ControlLabel
                    bsClass="control-label"
                    srOnly={false}
                  >
                    Task Description
                  </ControlLabel>
                  <FormControl
                    bsClass="form-control"
                    componentClass="input"
                    onChange={[Function]}
                    placeholder="Enter description"
                    type="text"
                    value=""
                  />
                  <FormControlFeedback
                    bsClass="form-control-feedback"
                    bsRole="feedback"
                  />
                </FormGroup>
              </form>
            </ModalBody>,
            <ModalFooter
              bsClass="modal-footer"
              componentClass="div"
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Close
              </Button>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="primary"
                disabled={false}
                onClick={[Function]}
                type="submit"
              >
                Save
              </Button>
            </ModalFooter>,
          ],
          "dialogComponentClass": [Function],
          "enforceFocus": true,
          "keyboard": true,
          "manager": ModalManager {
            "add": [Function],
            "containers": Array [],
            "data": Array [],
            "handleContainerOverflow": true,
            "hideSiblingNodes": true,
            "isTopModal": [Function],
            "modals": Array [],
            "remove": [Function],
          },
          "onHide": [Function],
          "renderBackdrop": [Function],
          "restoreFocus": true,
          "show": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "modal-header",
              "children": <ModalTitle
                bsClass="modal-title"
                componentClass="h4"
              >
                New Task
              </ModalTitle>,
              "closeButton": false,
              "closeLabel": "Close",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "modal-title",
                "children": "New Task",
                "componentClass": "h4",
              },
              "ref": null,
              "rendered": "New Task",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "modal-body",
              "children": <form>
                <FormGroup
                  bsClass="form-group"
                  controlId="taskName"
                  validationState={null}
                >
                  <ControlLabel
                    bsClass="control-label"
                    srOnly={false}
                  >
                    Task Description
                  </ControlLabel>
                  <FormControl
                    bsClass="form-control"
                    componentClass="input"
                    onChange={[Function]}
                    placeholder="Enter description"
                    type="text"
                    value=""
                  />
                  <FormControlFeedback
                    bsClass="form-control-feedback"
                    bsRole="feedback"
                  />
                </FormGroup>
              </form>,
              "componentClass": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <FormGroup
                  bsClass="form-group"
                  controlId="taskName"
                  validationState={null}
                >
                  <ControlLabel
                    bsClass="control-label"
                    srOnly={false}
                  >
                    Task Description
                  </ControlLabel>
                  <FormControl
                    bsClass="form-control"
                    componentClass="input"
                    onChange={[Function]}
                    placeholder="Enter description"
                    type="text"
                    value=""
                  />
                  <FormControlFeedback
                    bsClass="form-control-feedback"
                    bsRole="feedback"
                  />
                </FormGroup>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-group",
                  "children": Array [
                    <ControlLabel
                      bsClass="control-label"
                      srOnly={false}
                    >
                      Task Description
                    </ControlLabel>,
                    <FormControl
                      bsClass="form-control"
                      componentClass="input"
                      onChange={[Function]}
                      placeholder="Enter description"
                      type="text"
                      value=""
                    />,
                    <FormControlFeedback
                      bsClass="form-control-feedback"
                      bsRole="feedback"
                    />,
                  ],
                  "controlId": "taskName",
                  "validationState": null,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "bsClass": "control-label",
                      "children": "Task Description",
                      "srOnly": false,
                    },
                    "ref": null,
                    "rendered": "Task Description",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "bsClass": "form-control",
                      "componentClass": "input",
                      "onChange": [Function],
                      "placeholder": "Enter description",
                      "type": "text",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "bsClass": "form-control-feedback",
                      "bsRole": "feedback",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": "form",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "modal-footer",
              "children": Array [
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="default"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Close
                </Button>,
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="primary"
                  disabled={false}
                  onClick={[Function]}
                  type="submit"
                >
                  Save
                </Button>,
              ],
              "componentClass": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "block": false,
                  "bsClass": "btn",
                  "bsStyle": "default",
                  "children": "Close",
                  "disabled": false,
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": "Close",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "block": false,
                  "bsClass": "btn",
                  "bsStyle": "primary",
                  "children": "Save",
                  "disabled": false,
                  "onClick": [Function],
                  "type": "submit",
                },
                "ref": null,
                "rendered": "Save",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
